finagle-thrift/src/main/scala/com/twitter/finagle/thrift/ThriftClientBufferedCodec.scala

```diff
@@ -26,7 +26,15 @@ object ThriftClientBufferedCodec {
   /**
    * Create a [[com.twitter.finagle.thrift.ThriftClientBufferedCodecFactory]]
    */
-  def apply() = new ThriftClientBufferedCodecFactory
+  def apply(): ThriftClientBufferedCodecFactory =
+    apply(Protocols.binaryFactory(), true)
+
+  def apply(protocolFactory: TProtocolFactory): ThriftClientBufferedCodecFactory =
+    apply(protocolFactory, true)
+
+  def apply(protocolFactory: TProtocolFactory,
```

Comment:
Oh, you're right. I was thinking we can add method to a client builder `attemptProtocolUpgrade`, but realized that it's Thrift-specific. We're planing to remove all the codecs from Finagle in the nearest future (we even have a ticket for that and it's planed to a next tech-debt sprint) so I'm not 100% sure we need to add new functionality there. Let me chat with people to see what they think. I'll get back to you. Thanks, this is a solid work!